src/components/Frame/components/Toast/Toast.tsx

```diff
@@ -90,9 +47,38 @@ export class Toast extends React.Component<ToastProps, never> {
       );
     }
 
-    this.clearDismissalTimeout();
-    if (onDismiss != null) {
-      this.timer = window.setTimeout(onDismiss, timeoutDuration);
-    }
-  }
+    timerRef.current = window.setTimeout(onDismiss, timeoutDuration);
+
+    return clearDismissalTimeout;
```

Comment:
Can remove the need for the timerRef entirely if we define the teardown function inline?

```js
timer = window.setTimeout(onDismiss, timeoutDuration);

return () => {
  window.clearTimeout(timer);
}